build/build.sh

```diff
@@ -49,7 +49,9 @@ echo "Building targets for ${ARCH}, generated targets in ${TARGETS_DIR} director
 
 echo "Building ${PKG}/cmd/nginx"
 
-git config --global --add safe.directory /go/src/k8s.io/ingress-nginx
+pushd /go/src/k8s.io/ingress-nginx
+git config --add safe.directory /go/src/k8s.io/ingress-nginx
```

Comment:
@strongjz did you add the `--global` flag just temporarily in your earlier PR on Friday.
I myself had removed it a while back because `--global` broke build on MacOS.
Asking to just get on the same page because although less likely to build controller in cloudbuild, in short future, not having the flag is still a blocker in main I suppose. So maybe you will add the flag back later !